src/picongpu/include/traits/UsesRNG.hpp

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright 2016 Marco Garten, Rene Widera
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <boost/mpl/bool.hpp>
+
+namespace picongpu
+{
+namespace traits
+{
+
+/** Checks if an object requires the RNG
+ *
+ * @tparam T_Object any object (class or typename)
+ *
+ * This struct must define
+ * ::type (boost::mpl::bool_<>)
+ */
+template<typename T_Object>
+struct UsesRNG
```

Comment:
I don't see the requirement you call an advantage. Boost happily works with meta-functions returning a value instead of a type. The `meta-function class` is formed by using a placeholder, so the apply-construct has a `::type` But I was not able to find any requirement to have the function itself have a type member instead of a class. The following works fine:

    #include <boost/mpl/vector.hpp>
    #include <boost/mpl/copy_if.hpp>
    #include <boost/type_traits.hpp>

    namespace bmpl = boost::mpl;

    template<class T>
    struct Foo{ enum{value=true};};

    template<>
    struct Foo<int>{ enum{value=false};};

    typedef bmpl::vector<int, unsigned> MyVec;

    typedef bmpl::copy_if<MyVec, boost::is_unsigned<bmpl::_1> > Result;
    typedef bmpl::copy_if<MyVec, Foo<bmpl::_1> > Result2;
    typedef bmpl::copy_if<MyVec, bmpl::and_<Foo<bmpl::_1>, boost::is_unsigned<bmpl::_1> > > Result3;

    int main(){}

So IMO using a type for boolean traits just complicates things and distracts from the very nature of the trait.